lifecycle-operator/webhooks/pod_mutator/handlers/pod_annotation_test.go

```diff
@@ -0,0 +1,498 @@
+package handlers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/testr"
+	apicommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/apis/lifecycle/v1alpha3/common"
+	fakeclient "github.com/keptn/lifecycle-toolkit/lifecycle-operator/controllers/common/fake"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const appname = "SOME-APP-NAME"
+const lowerAppName = "some-app-name"
+const workloadName = "some-workload-name"
+const preDep = "some-pre-deployment-task"
+const postDep = "some-post-deployment-task"
+const preEval = "some-pre-deployment-evaluation"
+const postEval = "some-post-deployment-evaluation"
+const version = "v1.0.0"
+const uid = "this-is-the-pod-uid"
+
+func Test_copyAnnotationsIfParentAnnotated(t *testing.T) {
+	testNamespace := "test-namespace"
+	rsUidWithDpOwner := types.UID("this-is-the-replicaset-with-dp-owner")
+	rsUidWithNoOwner := types.UID("this-is-the-replicaset-with-no-owner")
+	testStsUid := types.UID("this-is-the-stateful-set-uid")
+	tstStsName := "test-stateful-set"
+	testDsUid := types.UID("this-is-the-daemon-set-uid")
+	testDsName := "test-daemon-set"
+
+	rsWithDpOwner := &appsv1.ReplicaSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "ReplicaSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-replicaset1",
+			UID:       rsUidWithDpOwner,
+			Namespace: testNamespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					Kind: "Deployment",
+					Name: "this-is-the-deployment",
+					UID:  "this-is-the-deployment-uid",
+				},
+			},
+		},
+	}
+	// TODO: fix tests where an RS has a STS or DS as owner. they should not have a RS in between
```

Comment:
I do not understand this comment from @mowies, not sure what test is needed here

Reply:
i really don't remember anymore... it's probably irrelevant by now

Reply:
I will get rid of it then 😄